src/ucc/api/ucc.h

```diff
@@ -635,15 +635,18 @@ enum ucc_context_attr_field {
  *
  * @brief OOB collective operation for creating the context
  */
-typedef struct ucc_context_oob_coll {
+typedef struct ucc_oob_coll {
     ucc_status_t    (*allgather)(void *src_buf, void *recv_buf, size_t size,
                                  void *allgather_info,  void **request);
     ucc_status_t    (*req_test)(void *request);
     ucc_status_t    (*req_free)(void *request);
     uint32_t        participants;
     void            *coll_info;
-}  ucc_context_oob_coll_t;
+    uint32_t        rank;
+}  ucc_oob_coll_t;
 
+typedef ucc_oob_coll_t ucc_context_oob_coll_t;
+typedef ucc_oob_coll_t ucc_team_oob_coll_t;
```

Comment:
can't do that because then the def of the whole struct ucc_oob_coll_t should go to def as well. Maybe remove team/context oob altogether, keeping just ucc_oob_coll_t? @manjugv